commit 584abd5d73b0bba6cc77c30191085c1ad794b0b5 (refs/changes/49/1049/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2016-12-02T20:27:11+01:00 (2 years, 10 months ago)
    
    MANTA-2999 muskie: Upgrade Node.js version from 0.10 to 4.x (or 6.x) before its EOL in Oct 2016

diff --git a/Makefile b/Makefile
index 64b5393..df8b787 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2016, Joyent, Inc.
 #
 
 #
@@ -55,8 +55,8 @@ CLEAN_FILES += node_modules
 #
 NAME 			= muskie
 NODE_PREBUILT_TAG       = zone
-NODE_PREBUILT_VERSION	:= v0.10.48
-NODE_PREBUILT_IMAGE     = fd2cc906-8938-11e3-beab-4359c665ac99
+NODE_PREBUILT_VERSION	:= v4.6.1
+NODE_PREBUILT_IMAGE     = 18b094b0-eb01-11e5-80c1-175dac7ddf02
 
 include ./tools/mk/Makefile.defs
 include ./tools/mk/Makefile.node_prebuilt.defs
diff --git a/etc/config.coal.json b/etc/config.coal.json
index 5465d66..3f15766 100644
--- a/etc/config.coal.json
+++ b/etc/config.coal.json
@@ -21,7 +21,10 @@
     "moray": {
         "connectTimeout": 2000,
         "host": "electric-moray.coal.joyent.us",
-        "port": 2020
+        "port": 2020,
+        "retry": {
+            "retries": 5
+        }
     },
     "marlin": {
         "connectTimeout": 2000,
@@ -31,6 +34,23 @@
             "expiry": 30
         }
     },
+    "cueballHttpAgent": {
+        "resolver": ["nameservice.coal.joyent.us"],
+        "initialDomains": [
+            "authcache.coal.joyent.us"
+        ],
+        "spares": 4,
+        "maximum": 20,
+        "recovery": {
+            "default": {
+                "timeout": 2000,
+                "maxTimeout": 10000,
+                "retries": 5,
+                "delay": 250,
+                "maxDelay": 2000
+            }
+        }
+    },
     "medusa": {
         "moray": {
             "host": "electric-moray.coal.joyent.us",
@@ -45,7 +65,8 @@
         "connectTimeout": 2000,
         "lag": 86400,
         "url": "tcp://1.moray.coal.joyent.us:2020",
-        "multiDC": false
+        "multiDC": false,
+        "ignoreSize": true
     },
     "sharkConfig": {
         "connectTimeout": 2000,
diff --git a/lib/common.js b/lib/common.js
index 7ee2a83..65f5ed8 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 var EventEmitter = require('events').EventEmitter;
@@ -851,13 +851,6 @@ module.exports = {
         function setup(req, res, next) {
             req.config = options;
             req.moray = options.moray();
-
-            // MANTA-331: while a trailing '/' is ok in HTTP,
-            // this messes with the consistent hashing, so
-            // ensure there isn't one
-            /* JSSTYLED */
-            req._path = req._path.replace(/\/*$/, '');
-
             req.jobCache = options.jobCache;
 
             req.log = (req.log || options.log).child({
diff --git a/lib/medusa/connector.js b/lib/medusa/connector.js
index 35ab6ad..141a3aa 100644
--- a/lib/medusa/connector.js
+++ b/lib/medusa/connector.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -54,6 +54,11 @@ function MedusaConnector(options) {
     self.moray = moray.createClient({
         log: options.log,
         connectTimeout: self.connectTimeout,
+        reconnect: true,
+        retry: { // try to reconnect forever
+            maxTimeout: 30000,
+            retries: 5
+        },
         host: options.moray.host,
         port: options.moray.port
     });
diff --git a/lib/picker.js b/lib/picker.js
index f70f6c3..29de1f0 100644
--- a/lib/picker.js
+++ b/lib/picker.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 //
@@ -316,6 +316,11 @@ function Picker(opts) {
     this.client = moray.createClient({
         log: opts.log,
         connectTimeout: opts.connectTimeout,
+        reconnect: true,
+        retry: {
+            maxTimeout: 30000,
+            retries: 5
+        },
         url: opts.url
     });
     this.connectTimeout = opts.connectTimeout || 1000;
@@ -394,7 +399,6 @@ Picker.prototype.choose = function choose(opts, cb) {
         }
     });
     dcs = shuffle(dcs);
-
     if ((replicas > 1 && this.multiDC && dcs.length < 2) ||
         !dcs.length ||
         (replicas > 1 && dcs.some(function (dc) {
diff --git a/lib/server.js b/lib/server.js
index 1bf8b7e..0af585b 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 var crypto = require('crypto');
@@ -44,7 +44,7 @@ var JOBS_LIVE_RE = /(state|status|name)=/i;
 ///--- Helpers
 
 // Always force JSON
-function formatJSON(req, res, body) {
+function formatJSON(req, res, body, callback) {
     if (body instanceof Error) {
         body = translateError(body, req);
         res.statusCode = body.statusCode || 500;
@@ -76,7 +76,7 @@ function formatJSON(req, res, body) {
     res.setHeader('Content-MD5', md5);
     res.setHeader('Content-Type', 'application/json');
 
-    return (data);
+    return callback(null, data);
 }
 
 
@@ -166,7 +166,7 @@ function createServer(options, clearProxy) {
     server.pre(function routeLiveJobs(req, res, next) {
         var tmp = JOBS_ROOT_RE.exec(req.path());
         if (tmp && tmp.length > 1 && JOBS_LIVE_RE.test(req.query()))
-            req._path = '/' + tmp[1] + '/jobs/live';
+            req.url = url.parse('/' + tmp[1] + '/jobs/live');
 
         next();
     });
@@ -176,7 +176,7 @@ function createServer(options, clearProxy) {
 
     server.use(common.earlySetupHandler(options));
     server.use(restify.dateParser(options.maxRequestAge || 300));
-    server.use(restify.queryParser());
+    server.use(restify.queryParser({allowDots: false, plainObjects: false}));
     server.use(common.authorizationParser);
     server.use(auth.checkIfPresigned);
     server.use(common.enforceSSLHandler(options));
diff --git a/lib/shark_client.js b/lib/shark_client.js
index 7a15947..d904066 100644
--- a/lib/shark_client.js
+++ b/lib/shark_client.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 var EventEmitter = require('events').EventEmitter;
@@ -55,7 +55,6 @@ util.inherits(SharkResponseError, Error);
 
 function _request(opts, cb) {
     cb = once(cb);
-
     var req = http.request(opts);
     /*
      * This timer represents the timeout for connecting to the shark
@@ -126,13 +125,20 @@ function request(thisp, method, opts, cb) {
     cb = once(cb);
 
     var log = thisp.log;
+
+    var headers = {
+        connection: 'keep-alive',
+        'x-request-id': opts.requestId
+    };
+    // Core HTTP Request "setHeader" method will throw an exception when
+    // trying to set a header to an undefined value, avoid it:
+    if (opts.range !== undefined) {
+        headers.range = opts.range;
+    }
+
     var _opts = {
         connectTimeout: thisp.connectTimeout,
-        headers: opts.headers || {
-            connection: 'keep-alive',
-            'x-request-id': opts.requestId,
-            range: opts.range
-        },
+        headers: opts.headers || headers,
         hostname: thisp.hostname,
         method: method,
         path: '/' + (opts.creator || opts.owner) + '/' + opts.objectId,
diff --git a/package.json b/package.json
index 30922b1..b9ecedd 100644
--- a/package.json
+++ b/package.json
@@ -6,40 +6,40 @@
     "private": true,
     "repository": {
         "type": "git",
-        "url": "git+ssh://git@github.com:joyent/muskie.git"
+        "url": "https://github.com/joyent/muskie.git"
     },
     "dependencies": {
-        "aperture-config": "git+ssh://git@github.com:joyent/aperture-config.git#master",
-        "assert-plus": "0.1.5",
+        "aperture-config": "https://github.com/joyent/aperture-config.git#master",
+        "assert-plus": "1.0.0",
         "backoff": "2.3.0",
-        "bunyan": "0.22.1",
-        "bunyan-syslog": "0.2.2",
+        "bunyan": "1.8.1",
+        "bunyan-syslog": "0.3.1",
         "cueball": "1.1.1",
         "dashdash": "1.3.2",
         "deep-equal": "0.0.0",
-        "dtrace-provider": "0.2.8",
-        "http-signature": "1.1.0",
+        "dtrace-provider": "0.8.0",
+        "http-signature": "1.1.1",
         "keep-alive-agent": "0.0.1",
-        "keyapi": "git+ssh://git@github.com:joyent/keyapi.git#c30dd2710ad2175095dc0e96479686fa774b8063",
-        "libmanta": "git+ssh://git@github.com:joyent/node-libmanta.git#master",
-        "libuuid": "0.1.2",
+        "keyapi": "https://github.com/joyent/keyapi.git#e14b3d582e1d9d338b7082d61f34ba8d1bbc540a",
+        "libmanta": "https://github.com/joyent/node-libmanta.git#MANTA-3031",
+        "libuuid": "0.2.1",
         "lru-cache": "2.3.1",
         "lstream": "0.0.4",
         "mahi": "2.0.1",
-        "marlin": "git+ssh://git@github.com:joyent/manta-marlin.git#master",
+        "marlin": "https://github.com/joyent/manta-marlin.git#dev-MANTA-3039",
         "mime": "1.2.11",
         "moray": "^2.0.0",
         "once": "1.3.0",
-        "restify": "2.6.3",
-        "vasync": "1.4.3",
         "verror": "^1.7.0",
+        "restify": "4.1.1",
+        "vasync": "1.6.4",
         "watershed": "0.3.0",
         "xtend": "2.1.1"
     },
     "devDependencies": {
-        "smartdc": "git+ssh://git@github.com:joyent/node-smartdc.git#v7.3",
-        "manta": "1.4.5",
-        "nodeunit": "0.8.6",
+        "smartdc": "8.1.0",
+        "manta": "4.0.0",
+        "nodeunit": "0.10.2",
         "node-uuid": "1.4.1"
     },
     "scripts": {
